instrumentation/apache-httpclient/apache-httpclient-4.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachehttpclient/v4_0/ApacheHttpClientInstrumentationModule.java

```diff
@@ -198,9 +192,7 @@ public static void methodEnter(
       scope = context.makeCurrent();
 
       // Wrap the handler so we capture the status code
-      if (handler instanceof ResponseHandler) {
-        handler = new WrappingStatusSettingResponseHandler(context, (ResponseHandler<?>) handler);
-      }
+      handler = new WrappingStatusSettingResponseHandler<>(context, handler);
```

Comment:
It might be better to check `handler` for `null` before wrapping to ensure that in case of `null` handler execute would fail when it checks args passed to the method instead of failing with `NullPointerException` in `WrappingStatusSettingResponseHandler.handleResponse`

Reply:
thanks! fixed